File: apps/bff/src/store/store.context.ts
import { defaultStoreCode, stores } from "@commerce/store-config";
import { Inject, Injectable, Scope } from "@nestjs/common";
import { REQUEST } from "@nestjs/core";
import { STORE_CONTEXT_KEY } from "./store.middleware";

const defaults = stores[defaultStoreCode]!;

@Injectable({ scope: Scope.REQUEST })
export class StoreContext {
  constructor(@Inject(REQUEST) private readonly request: any) {}

  get storeCode(): string {
    return this.request?.[STORE_CONTEXT_KEY]?.storeCode ?? defaults.storeCode;
  }

  get locale(): string {
    return this.request?.[STORE_CONTEXT_KEY]?.locale ?? defaults.locale;
  }

  get currency(): string {
    return this.request?.[STORE_CONTEXT_KEY]?.currency ?? defaults.currency;
  }

  get customerId(): string | undefined {
    return this.request?.[STORE_CONTEXT_KEY]?.customerId;
  }

  get isAuthenticated(): boolean {
    return this.customerId !== undefined;
  }
}

File: apps/api/src/main/java/com/commerce/api/config/StoreContext.java
package com.commerce.api.config;

/**
 * Holds per-request store and locale context, extracted from HTTP headers.
 */
public record StoreContext(String storeCode, String locale) {

    public static final String DEFAULT_STORE_CODE = "fr";
    public static final String DEFAULT_LOCALE = "fr-FR";

    public String language() {
        if (locale == null || !locale.contains("-")) {
            return "fr";
        }
        return locale.split("-")[0];
    }
}
